client/components/marketing-survey/cancel-purchase-form/cancellation-reasons.ts

```diff
@@ -0,0 +1,285 @@
+import { JETPACK_PRODUCTS_LIST } from '@automattic/calypso-products';
+import { translate } from 'i18n-calypso';
+import { DOWNGRADEABLE_PLANS_FROM_PLAN } from 'calypso/my-sites/plans/jetpack-plans/constants';
+import slugToSelectorProduct from 'calypso/my-sites/plans/jetpack-plans/slug-to-selector-product';
+import type { SelectorProduct } from 'calypso/my-sites/plans/jetpack-plans/types';
+import type { ReactChild } from 'react';
+
+type TranslateReturnType = ReactChild | string;
+
+export interface CancellationReasonBase {
+	/**
+	 * A string value that will be reported.
+	 */
+	value: string;
+
+	/**
+	 * A string that will be displayed to the user.
+	 */
+	label: TranslateReturnType;
+
+	/**
+	 * Whether this option is disabled.
+	 */
+	disabled?: boolean;
+}
+
+export interface CancellationReason extends CancellationReasonBase {
+	/**
+	 * placeholder text for the additional input
+	 */
+	textPlaceholder?: TranslateReturnType;
+
+	/**
+	 * Default value for the sub category select
+	 */
+	selectInitialValue?: string;
+
+	/**
+	 * Default label for the sub category select
+	 */
+	selectLabel?: TranslateReturnType;
+
+	/**
+	 * Options for the sub category select
+	 */
+	selectOptions?: CancellationReasonBase[];
+}
+
+/**
+ * The reason always shown as the first option.
+ */
+export const PLACEHOLDER: CancellationReason = {
+	value: '',
+	label: translate( 'Select your reason' ),
+	disabled: true,
+};
+
+/**
+ * The reason always shown at the end.
+ */
+export const LAST_REASON: CancellationReason = {
+	value: 'anotherReasonOne',
+	label: translate( 'Another reason…' ),
+	textPlaceholder: translate( 'Can you please specify?' ),
+};
+
+export const CANCELLATION_REASONS: CancellationReason[] = [
+	{
+		value: 'price/budget',
+		label: translate( 'Price/Budget' ),
+		selectOptions: [
+			PLACEHOLDER,
+			{
+				value: 'tooExpensice',
+				label: translate( 'Too expensive' ),
+			},
+			{
+				value: 'wantCheaperPlan',
+				label: translate( 'I want a cheaper plan' ),
+			},
+			{
+				value: 'freeIsGoodEnough',
+				label: translate( 'Free is good enough for me' ),
+			},
+		],
+	},
+	{
+		value: 'couldNotFinish',
+		label: translate( 'Couldn’t finish my site' ),
+		selectOptions: [
+			PLACEHOLDER,
+			{
+				value: 'noTime',
+				label: translate( 'I don’t have time.' ),
+			},
+			{
+				value: 'needProfessionalHelp',
+				label: translate( 'Need professional help to build my site' ),
+			},
+			{
+				value: 'siteIsNotReady',
+				label: translate( 'My site is not ready' ),
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'My site is not ready.' )` *ES Score: 9*